src/drivers/sifive,spi0.c

```diff
@@ -122,43 +140,81 @@ int __metal_driver_sifive_spi0_transfer(struct metal_spi *gspi,
                                       char *tx_buf,
                                       char *rx_buf)
 {
+    /* Transmit and receive buffer must not be empty */
+    if (tx_buf == NULL || rx_buf == NULL) {
+        return 1;
+    }
```

Comment:
I agree with the `rx_buf == NULL` part. However, about the `tx_buf  == null` situation, is the user just going to send `len` bytes of 0's? That is treated as dummy bytes by most spi devices.

Reply:
Actually you require the ability to "stuff 0" or "stuff 1" - this might be something you place in the "config" structure parameter to this function, add an element  "int stuffing_value" to the config struct.